Move from Microsoft.Bcl.Json.Sources to System.Text.Json

dougbu opened the pull request:
- fix downgrade of CoreFx packages in core-sdk
- remove System.Runtime.CompilerServices.Unsafe references from affected projects in hopes they're no longer needed

details:
- CoreFx no longer produces the Microsoft.Bcl.Json.Sources package
- System.Text.Json is now .NET Standard 2.0-compatible
- see also dotnet/corefx#37129 and aspnet/Extensions#1610

dougbu committed:
Move from Microsoft.Bcl.Json.Sources to System.Text.Json
- fix downgrade of CoreFx packages in core-sdk
- remove System.Runtime.CompilerServices.Unsafe references from affected projects in hopes they're no longer needed

details:
- CoreFx no longer produces the Microsoft.Bcl.Json.Sources package
- System.Text.Json is now .NET Standard 2.0-compatible
- see also dotnet/corefx#37129 and aspnet/Extensions#1610

BrennanConroy commented:
You missed the other projects that use System.Text.Json, like MVC and Security

natemcmaster commented:
You'll need to add this assembly to this section to resolve the error in ResolveReferences.targets.

https://github.com/aspnet/AspNetCore/blob/ba346b45a22295955787477f6869f310aa62e224/eng/SharedFramework.External.props#L84-L92

dougbu committed:
!fixup! to satisfy various systems
- change eng/SharedFramework.Externsl.props
- remove System.Buffers references too; System.Text.Json brings this in transitively
  - suppress baseline warning about System.Buffers
- remove special case in Microsoft.AspNetCore.SignalR.Protocols.Json project
  - Microsoft.AspNetCore.SignalR.Common brings necessary .NET Standard 2.0 packages in transitively
- regenerate ref project files

BrennanConroy commented on src/SignalR/common/SignalR.Common/ref/Microsoft.AspNetCore.SignalR.Common.csproj:
```diff
@@ -5,12 +5,11 @@
-    <Reference Include="Microsoft.Extensions.Options" />
-    <Reference Include="System.Buffers" />
-    <Reference Include="System.Text.Json" />
+    <Reference Include="Microsoft.AspNetCore.Connections.Abstractions"  />
```
Whats with all the extra spaces in the references?

dougbu replied:
That's just the way the generator works and it's not our generator. In any case GenApi isn't about pretty files 😺

BrennanConroy replied:
Oh it's the ref file, carry on 😄

natemcmaster commented on src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj:
```diff
@@ -25,9 +25,10 @@
+    <Reference Include="System.Text.Json" />
+
+    <!-- System.Text.Json brings System.Buffers in transitively. No need for both here. -->
+    <SuppressBaselineReference Include="System.Buffers" />
```
Can you move this into the block below? This will prevent suppressions from bleeding over into the next major or minor version.

dougbu replied:
I'll try it.

natemcmaster commented on src/SignalR/common/Http.Connections.Common/src/Microsoft.AspNetCore.Http.Connections.Common.csproj:
```diff
@@ -20,9 +20,10 @@
+    <Reference Include="System.Text.Json" />
+
+    <!-- System.Text.Json brings System.Buffers in transitively. No need for both here. -->
+    <SuppressBaselineReference Include="System.Buffers" />
```
Same here.

natemcmaster reviewed (changes requested):
One nit about code organization. Otherwise looks good.

dougbu committed:
Address PR comment: Suppress baseline reference based on ASP.NET Core version (not TFM)

dougbu commented:
:up::date: for @natemcmaster

dougbu committed:
Merge branch 'master' into dougbu/no.sources.package
